TaskClass.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Enum representing possible task states
+
+public enum TaskStatus
+{
+	NotStarted,
+	InProgress,
+	Completed,
+	OnHold       // If some issue prevents progression on a task
+}
+
+// Enum representing priotity level of a task
+
+public enum TaskPriority
+{
+	Low,
+	Medium,
+	High,
+	Critical
+}
+
+public class Task
+{
+	// Properties for a task object
+
+	public string TaskName { get; private set; }
+	public string Description { get; private set; }
+	public TaskStatus Status { get; private set; }
+	public TaskPriority Priority { get; private set; }
+	public DateTime CreatedOn { get; private set; } // Timestamping the creation of a task
+	public DateTime? CompletedOn { get; private set; } // also the completion date
+	public DateTime? DueDate { get; private set; }  // Optional parameter for a deadline
+	public string IsAssignedTo { get; private set; }
+
+
+	// Constructor for intializing new tasks
+
+	public Task(string taskName, string description, TaskPriority priority, DateTime? dueDate = null) // default value set to null since this parameter is optional
+	{
+		if (string.IsNullOrEmpty(taskName)) // Validation for task name
+			throw new ArgumentNullException("Task must have a name");
+
+		// Properties initialized
+
+		TaskName = taskName;
+		Description = description;
+		Priority = priority;
+		Status = TaskStatus.NotStarted; // A new task hasn't been started as default
+		CreatedOn = DateTime.Now; // To record creation time
+		DueDate = dueDate;
+	}
+
+	// Method for changing task status
+
+	public void ChangeTaskStatus(TaskStatus newStatus)
+	{
+		if (Status == TaskStatus.Completed) // Finished task's shouldn't be modified
```

Comment:
Why not? What if you accidently set it as completed?

Reply:
A good point, removed this check